src/components/QuranReader/TafsirView/TafsirBody.tsx

```diff
@@ -0,0 +1,211 @@
+/* eslint-disable max-lines */
+/* eslint-disable i18next/no-literal-string */
+import { useCallback, useEffect, useState } from 'react';
+
+import classNames from 'classnames';
+import uniq from 'lodash/uniq';
+import useTranslation from 'next-translate/useTranslation';
+import { useSelector, shallowEqual, useDispatch } from 'react-redux';
+import useSWR from 'swr/immutable';
+
+import SurahAndAyahLanguageSelection from './SurahAndAyahSelection';
+import TafsirGroupMessage from './TafsirGroupMessage';
+import TafsirAndLanguageSelection from './TafsirSelection';
+import TafsirSkeleton from './TafsirSkeleton';
+import styles from './TafsirView.module.scss';
+
+import { fetcher } from 'src/api';
+import DataFetcher from 'src/components/DataFetcher';
+import Separator from 'src/components/dls/Separator/Separator';
+import VerseText from 'src/components/Verse/VerseText';
+import { selectQuranReaderStyles } from 'src/redux/slices/QuranReader/styles';
+import { selectSelectedTafsirs, setSelectedTafsirs } from 'src/redux/slices/QuranReader/tafsirs';
+import QuranReaderStyles from 'src/redux/types/QuranReaderStyles';
+import { getDefaultWordFields } from 'src/utils/api';
+import { makeTafsirContentUrl, makeTafsirsUrl } from 'src/utils/apiPaths';
+import { areArraysEqual } from 'src/utils/array';
+import { fakeNavigate, getVerseTafsirNavigationUrl } from 'src/utils/navigation';
+import { getFirstAndLastFirstKeys, getVerseWords, makeVerseKey } from 'src/utils/verse';
+import { TafsirsResponse } from 'types/ApiResponses';
+
+type TafsirBodyProps = {
+  initialChapterId: string;
+  initialVerseNumber: string;
+  initialTafsirData?: TafsirsResponse;
+  initialTafsirId?: number;
+};
+
+const TafsirBody = ({
+  initialChapterId,
+  initialVerseNumber,
+  initialTafsirData,
+  initialTafsirId,
+}: TafsirBodyProps) => {
+  const dispatch = useDispatch();
+  const quranReaderStyles = useSelector(selectQuranReaderStyles, shallowEqual) as QuranReaderStyles;
+  const { lang } = useTranslation();
+  const userPreferredTafsirIds = useSelector(selectSelectedTafsirs, areArraysEqual);
+
+  const [selectedChapterId, setSelectedChapterId] = useState(initialChapterId);
+  const [selectedVerseNumber, setSelectedVerseNumber] = useState(initialVerseNumber);
+  const [selectedLanguage, setSelectedLanguage] = useState(lang);
```

Comment:
The selected language should be coming from the selected tafsir, not the user locale.